setup.py

```diff
@@ -19,6 +24,189 @@
+            if IS_LINUX or IS_DARWIN:
+                CUDA_HOME = os.getenv('CUDA_HOME', LINUX_HOME)
+            else:
+                CUDA_HOME = os.getenv('CUDA_PATH', '').replace('\\', '/')
+                if CUDA_HOME == '' and len(WINDOWS_HOME) > 0:
+                    CUDA_HOME = WINDOWS_HOME[0].replace('\\', '/')
+            if not os.path.exists(CUDA_HOME):
+                # We use nvcc path on Linux and cudart path on macOS
+                if IS_LINUX or IS_WINDOWS:
+                    cuda_path = find_nvcc()
+                else:
+                    cudart_path = ctypes.util.find_library('cudart')
+                    if cudart_path is not None:
+                        cuda_path = os.path.dirname(cudart_path)
+                    else:
+                        cuda_path = None
+                if cuda_path is not None:
+                    CUDA_HOME = os.path.dirname(cuda_path)
+                else:
+                    CUDA_HOME = None
+            cuda_version = find_cuda_version(CUDA_HOME)
+        return cuda_version
+
+    def get_gcc():
+        proc = subprocess.Popen(['gcc', '--version'],
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        out, err = proc.communicate()
+        gcc_version = re.search(r'[0-9]+\.[0-9]+\.[0-9]+', out.decode())
+        if gcc_version:
+            return gcc_version.group()
+        else:
+            return None
+
+    def _minimal_ext_cmd(cmd):
+        # construct minimal environment
+        env = {}
+        for k in ['SYSTEMROOT', 'PATH', 'HOME']:
+            v = os.environ.get(k)
+            if v is not None:
+                env[k] = v
+        # LANGUAGE is used on win32
+        env['LANGUAGE'] = 'C'
+        env['LANG'] = 'C'
+        env['LC_ALL'] = 'C'
+        out = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, env=env).communicate()[0]
+        return out
+
+    def get_git_hash(fallback='unknown', digits=None):
+        """Get the git hash of the current repo.
+
+        Args:
+            fallback (str, optional): The fallback string when git hash is
+                unavailable. Defaults to 'unknown'.
+            digits (int, optional): kept digits of the hash. Defaults to None,
+                meaning all digits are kept.
+
+        Returns:
+            str: Git commit hash.
+        """
+
+        if digits is not None and not isinstance(digits, int):
+            raise TypeError('digits must be None or an integer')
+
+        try:
+            out = _minimal_ext_cmd(['git', 'rev-parse', 'HEAD'])
+            sha = out.strip().decode('ascii')
+            if digits is not None:
+                sha = sha[:digits]
+        except OSError:
+            sha = fallback
+
+        return sha
+
+    mmcv_root = Path(__file__).parent
+    version_path = mmcv_root / 'mmcv' / 'version.py'
+    gcc_version = get_gcc()
+    cuda_version = get_cuda()
+    git_hash = get_git_hash()
```

Comment:
```suggestion
    git_hash = get_git_hash(digits=7)
```
We just need to keep the first 7 characters.

Reply:
done